internal/pkg/scaffold/controller_kind.go

```diff
@@ -26,39 +30,112 @@ type ControllerKind struct {
 
 	// Resource defines the inputs for the controller's primary resource
 	Resource *Resource
+	// CustomImport holds the import path for a built-in or custom Kubernetes
+	// API that this controller reconciles, if specified by the scaffold invoker.
+	CustomImport string
+
+	// ImportMap maps all imports destined for the scaffold to their import
+	// identifier, if any.
+	ImportMap map[string]string
+	// GoImportIdent is the import identifier for the API reconciled by this
+	// controller.
+	GoImportIdent string
 }
 
-func (s *ControllerKind) GetInput() (input.Input, error) {
+func (s *ControllerKind) GetInput() (_ input.Input, err error) {
 	if s.Path == "" {
 		fileName := s.Resource.LowerKind + "_controller.go"
 		s.Path = filepath.Join(ControllerDir, s.Resource.LowerKind, fileName)
 	}
 	// Error if this file exists.
 	s.IfExistsAction = input.Error
 	s.TemplateBody = controllerKindTemplate
+
+	// Set imports.
+	s.ImportMap = controllerKindImports
```

Comment:
Nit: This function is getting long, can we extract out the import handling into its own function? If possible of course.